### ISSUE TITLE
make dvc repro STAGE running all parametrized stages

### ISSUE THREAD

**Issue by johnnychen94** [iterative/dvc#4958]
```yaml
stages:
  build:
    foreach:
      - 15
      - 25
      - 35
    do:
      cmd: echo ${item}
```

```console
jc@mathai3:~/Downloads/dvc_test$ dvc repro build@15
Running callback stage 'build@15' with command:
	echo 15
15
Use `dvc push` to send your updates to remote storage.
```

The following version doesn't work, even though `build` is defined in `stages` section.

```console
jc@mathai3:~/Downloads/dvc_test$ dvc repro build
ERROR: "Stage 'build' not found inside 'dvc.yaml' file"
```

I wish we could make it equivalent to

```console
for item in 15 25 35; do
    dvc repro build@$item
done
```

Furthermore, depends on whether there's a lock, multiple stages can be run in parallel.

**Comment by skshetry** [comment]
Hi, @johnnychen94. Glad to see an issue for the parameterization (and, seeing someone using it :smile:). I am working on this feature and will be implemented around the pre-release (which might be backwards-incompatible).

**Comment by skshetry** [comment]
@johnnychen94, I have created #4976 to implement this. Your feedback would be greatly appreciated.

**Comment by johnnychen94** [comment]
It's not closed related so I put my thoughts in #4979 on what I'm expecting.
